openverse_catalog/dags/providers/provider_api_scripts/science_museum.py

```diff
@@ -45,198 +27,202 @@
-        response = delay_request.get(endpoint, query_param, headers=headers)
-        try:
-            response_json = response.json()
-            if "data" in response_json.keys():
-                data = response_json.get("data")
-                break
-        except Exception as e:
-            logger.error(f"Failed to due to {e}")
-    return data
-
-
-def _handle_object_data(batch_data):
-    image_count = 0
-    for obj_ in batch_data:
-        id_ = obj_.get("id")
-        if id_ in RECORD_IDS:
-            continue
-        RECORD_IDS.append(id_)
-        foreign_landing_url = obj_.get("links", {}).get("self")
+            # Increment `skip` by the batch limit.
+            return {
+                **old_query_params,
+                "date[from]": from_,
+                "date[to]": to_,
+                "page[number]": old_query_params["page[number]"] + 1,
+            }
+
+    def get_media_type(self, record):
+        # This provider only supports Images.
+        return "image"
+
+    def get_batch_data(self, response_json):
+        if response_json:
+            return response_json.get("data")
+        return None
+
+    def get_record_data(self, record):
+        id_ = record.get("id")
+        if id_ in ScienceMuseumDataIngester.RECORD_IDS:
+            return None
+        ScienceMuseumDataIngester.RECORD_IDS.append(id_)
+        foreign_landing_url = record.get("links", {}).get("self")
         if foreign_landing_url is None:
-            continue
-        obj_attributes = obj_.get("attributes")
-        if obj_attributes is None:
-            continue
-        title = obj_attributes.get("summary_title")
-        creator = _get_creator_info(obj_attributes)
-        metadata = _get_metadata(obj_attributes)
-        multimedia = obj_attributes.get("multimedia")
+            return None
+        attributes = record.get("attributes")
+        if attributes is None:
+            return None
+        title = attributes.get("summary_title")
+        creator = self._get_creator_info(attributes)
+
+        metadata = self._get_metadata(attributes)
+        multimedia = attributes.get("multimedia")
         if multimedia is None:
-            continue
+            return None
+        images = []
         for image_data in multimedia:
             foreign_id = image_data.get("admin", {}).get("uid")
             if foreign_id is None:
                 continue
             processed = image_data.get("processed")
-            source = image_data.get("source")
-            image_url, height, width = _get_image_info(processed)
+            (
+                image_url,
+                height,
+                width,
+                filetype,
+            ) = ScienceMuseumDataIngester._get_image_info(processed)
             if image_url is None:
                 continue
 
-            license_version = _get_license_version(source)
-            if license_version is None:
+            license_pair = self._get_license(image_data)
```

Comment:
Same note here as #600 - the `ScienceMuseumDataIngester` static method calls can just be `self`, for consistency with L108 as well.